src/zope/app/component/testing.py

```diff
@@ -11,185 +11,44 @@
 # FOR A PARTICULAR PURPOSE.
 #
 ##############################################################################
-"""Base Mix-in class for Placeful Setups
-
-Also contains common test related classes/functions/objects.
 """
+Test helpers.
+"""
+from __future__ import absolute_import
 
-import os
 import zope.interface
 import zope.site.folder
-from zope.component.interfaces import IComponentLookup
-from zope.app.component.interfaces import ILocalSiteManager
-from zope.app.testing import setup
-from zope.app.testing.placelesssetup import PlacelessSetup
-
-from zope.app.testing.functional import ZCMLLayer
-from zope.traversing.api import traverse
-
-AppComponentLayer = ZCMLLayer(
-    os.path.join(os.path.split(__file__)[0], 'ftesting.zcml'),
-    __name__, 'AppComponentLayer', allow_teardown=True)
-
-class Place(object):
-
-    def __init__(self, path):
-        self.path = path
-
-    def __get__(self, inst, cls=None):
-        if inst is None:
-            return self
-
-        try:
-            # Use __dict__ directly to avoid infinite recursion
-            root = inst.__dict__['rootFolder']
-        except KeyError:
-            root = inst.rootFolder = setup.buildSampleFolderTree()
-
-        return traverse(root, self.path)
-
-
-class PlacefulSetup(PlacelessSetup):
```

Comment:
It was simple to add back enough to testing to make z3c.jsontree happy while not adding in any deprecated dependencies and also maintaining test coverage. 

Whether z3c.jsontree should have a dependency on zope.app.component is a different question. It appears to only use it for testing one view (and it doesn't use zope.app.zcmlfiles), which only uses PlacefulSetup, so it's possible that dep could/should be easily broken anyway.

Implementing this 'PlacefulSetup' only requires zope.component, zope.container, zope.site and zope.traversing. Perhaps there's a better package to host it.